Sources/MobileSdk/MDocReaderBLEPeripheral.swift

```diff
@@ -113,18 +113,50 @@ class MDocReaderBLEPeripheral: NSObject {
             /// L2CAP flow.
             case .l2capRead: // We have a read on our L2CAP characteristic, start L2CAP flow.
                 machineState = .l2capAwaitChannelPublished
-                peripheralManager.publishL2CAPChannel(withEncryption: true)
+
+                // Setting the encryption flag here appears to break at least some Android devices; if `withEncryption`
+                // is set, when the Android device calls `.connect()` on the socket it throws a "resource not
+                // available" exception.  The data is already encrypted, so I'm setting it to false, and making Android
+                // recover from the exception and fall back to the old flow if necessary as well.  Support code for
+                // failover is in the next two states, below.
+
+                peripheralManager.publishL2CAPChannel(withEncryption: false)
                 update = true
 
             case .l2capAwaitChannelPublished:
                 if machinePendingState == .l2capChannelPublished {
                     machineState = machinePendingState
+                } else if machinePendingState == .stateSubscribed {
+
+                    // Failover case for Android; we could get this here, or in the published state.  Android devices
+                    // may be unable to connect() on an L2CAP socket (see the comment in .l2capRead, above), and if
+                    // they do they "switch tracks" to the old flow.  We need to notice that's happened and support
+                    // it here.
+
+                    print("Failover to non-L2CAP flow.")
```

Comment:
This is not a blocker and there may be a good reason for using `print` instead of `Logger` instance, but consider changing instances of `print` to use `os_log`, or other `Logger` type. 

https://developer.apple.com/documentation/os/logging/generating_log_messages_from_your_code

Reply:
There are other instances of `print` throughout the codebase, so this work might be better suited in another PR for logging.

Reply:
I chose `print()` in this instance because it was more about information useful during local debugging, which `os_log()` isn't ideal for.  `os_log()` goes (effectively) to `/var/log/syslog`, where it gets dumped into the maelstrom with all the other logs.  It's also subject to some limitations; it silently truncates log lines at 1024 bytes, for example, though that isn't relevant here.